router/flow.go

```diff
@@ -0,0 +1,50 @@
+package router
+
+import "net"
+
+// Just enough flow machinery for the weave router
+
+type MAC [6]byte
+
+func (mac MAC) String() string {
+	return net.HardwareAddr(mac[:]).String()
+}
+
+type PacketKey struct {
+	SrcMAC MAC
+	DstMAC MAC
+}
+
+type ForwardPacketKey struct {
+	SrcPeer *Peer
+	DstPeer *Peer
+	PacketKey
+}
+
+type FlowOp interface {
+	// The caller must supply an EthernetDecoder specific to this
+	// thread, which has already been used to decode the frame.
+	// The broadcast parameter is a hint whether the packet is
+	// being broadcast.
+	Send(frame []byte, dec *EthernetDecoder, broadcast bool)
+}
```

Comment:
Kind of.  Fast datapath introduces `FlowOps` that still represent a way to send a packet, but gain another interpretation as elements of ODP flow rules passed to the kernel.  That doesn't actually result in more methods being added to `FlowOp`, as not all `FlowOps` are fast datapath `FlowOps`.  But it requires the ability to test if a `FlowOp` implements another interface.  That wouldn't work with a function type.